src/libstore/path.hh

```diff
@@ -43,6 +44,19 @@ struct StorePath : rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>
         return !(*this == other);
     }
 
+    StorePath(StorePath && that) = default;
+
+    StorePath(const StorePath & that)
+    {
+        ffi_StorePath_clone_to(that, *this);
+    }
+
+    void operator = (const StorePath & that)
+    {
+        (rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>::operator = (that));
```

Comment:
Where does that magic size come from? I see that we start to duplicate that throughout (at least) this file a few times. Shouldn't there be a comment where the magic 24 comes from? For the `3 * sizeof(void *)` part one can probably guess that the struct hold 3 pointers. The 24 bytes could be 3 integers? Can't we use a single `sizeof(SomeStruct)` instead? Does this take into account alignment changes if a new member is added to whatever the underlying struct is?

I know you didn't introduce it but this might go out of sync with the other definition thus I am commenting.

Reply:
Nevermind the numbers, I am repeating the whole superclass. Maybe I should `#ifdef` and then `#undef` the superclass; that would help without leaking anything more to the files the include the header.